### ISSUE TITLE
Set explicit void return type in function builder if set by the user

### ISSUE THREAD

**Issue by Legion2** [outfoxx/typescriptpoet#24]
When the return type is explicitly set to void it should also be set in the generated code explicitly to void. When enabling strict type checking in typescript compiler with the `noImplicitAny` option, all functions without explicit return type default to any. 
Following code must be changed:
https://github.com/outfoxx/typescriptpoet/blob/62d7d4b276b6039bd04729ddb135a9428e61b03e/src/main/java/io/outfoxx/typescriptpoet/FunctionSpec.kt#L107-L109

**Comment by kdubb** [comment]
Agreed.

**Comment by Legion2** [comment]
I created #25

**Comment by tinacious** [comment]
I agree it would be great to add this to prevent `'functionName', which lacks return-type annotation, implicitly has an 'any' return type`.

It's also important to note that [`noImplicitAny`](https://www.typescriptlang.org/tsconfig/#noImplicitAny) is the default configuration when using strict mode:

<img width="999" alt="image" src="https://github.com/user-attachments/assets/c07c150d-39f7-4345-9904-2f98b9e27985" />


In case anyone finding this is interested in a workaround until the issue is addressed, I defined a type alias `Void` and used that as the value passed to `FunctionSpec#returns()`. It's kinda gross but it works. 🤷🏻‍♀️ Another option is to add TypeScript and ESLint disabling comments but that would just clear the error rather than enforce a void return type, so I opted for `Void`:

```kt
// Workaround for https://github.com/outfoxx/typescriptpoet/issues/24
val voidType = TypeAliasSpec
    .builder("Void", TypeName.VOID)
    .addTSDoc("Workaround for https://github.com/outfoxx/typescriptpoet/issues/24")
    .build()
fileSpecBuilder.addType(voidType)

val someInterface = InterfaceSpec
    .builder("MyInterface")
    .addFunction(
        FunctionSpec
            .builder("doSomething")
            .addModifiers(Modifier.ABSTRACT)
            .returns(TypeName.implicit("Void"))
            .build()
    )
    .build()
fileSpecBuilder.addType(someInterface)
```


Which outputs:

```ts
/**
 * Workaround for https://github.com/outfoxx/typescriptpoet/issues/24 */
type Void = void;

interface MyInterface {

  doSomething(): Void;

}
```

**Comment by kdubb** [comment]
@Legion2 @tinacious I've been quite preoccupied with other projects. I'll spend some time updating this project.

**Comment by kdubb** [comment]
That includes merging #25 😉
